pythonlib/src/mill/pythonlib/PythonModule.scala

```diff
@@ -21,59 +40,81 @@ trait PythonModule extends Module {
     PathRef(python)
   }
 
-  def typeCheck: T[Unit] = Task {
-    Task.traverse(moduleDeps)(_.typeCheck)()
-
-    os.call(
-      (pythonExe().path, "-m", "mypy", "--strict", sources().path),
-      stdout = os.Inherit,
-      cwd = T.workspace
+  // TODO: right now, any task that calls this helper will have its own python
+  // cache. This is slow. Look into sharing the cache between tasks.
```

Comment:
What we've done elsewhere is provide a `Task(persistent = true)` whose sole purpose is to return a `os.Path` for people to scribble over. Maybe we could do that here to provide the `PYTHONPYCACHEPREFIX`?

Reply:
If we have a task that returns a PathRef, then later we write to it after the task has run, wouldn't that mess with cache invalidation?

Reply:
It isn't really kosher but it works. But I think we can punt on it for this PR: populating the `__pycache__` isn't particularly slow so I doubt anyone will notice, and if it proves to be problematic we can optimize it later

Reply:
ok, I'll let it be as is for now then